### PR TITLE
targets can now inherit information from each other and show-command …

### PR DESCRIPTION
**Pull request by pboettch**
…implemented

Do not merge, only review and tests.

Documentation update is still needed.

Please also review @adri1mart1

### CODE REVIEW

File: chef/chef.py [hunk 1]
```diff
@@ -233,34 +299,34 @@ def update_directory(self, method, local_dir, has_remote, branch, rev):
...
-    def prepare_build_directory(self, path, target):
-        debug('Preparing directory:', path)
+    def prepare_build_directory(self, target):
+        debug('Preparing directory:', target.build_dir())
 
-        os.makedirs(path, exist_ok=True)
+        os.makedirs(target.build_dir(), exist_ok=True)
 
-        conf_path = os.path.join(path, 'conf')
+        conf_path = os.path.join(target.build_dir(), 'conf')
```
Comment (adri1mart1) [inline chef/chef.py:312]: I think this should be in the buildconfiguration class. 
```python3
def conf_dir(self):
    return os.path.join(self.build_dir(), 'conf')
```

File: chef/chef.py [hunk 2]
```diff
@@ -233,34 +299,34 @@ def update_directory(self, method, local_dir, has_remote, branch, rev):
...
         os.makedirs(conf_path, exist_ok=True)
 
-        dl_dir = '${TOPDIR}/' + os.path.relpath(self.config.dl_dir(), path)
+        dl_dir = '${TOPDIR}/' + os.path.relpath(self.config.dl_dir(), target.build_dir())
 
         sstate_dir = os.path.join(self.config.project_root(), 'sstate-cache')
-        sstate_dir = '${TOPDIR}/' + os.path.relpath(sstate_dir, path)
+        sstate_dir = '${TOPDIR}/' + os.path.relpath(sstate_dir, target.build_dir())
```
Comment (adri1mart1) [inline chef/chef.py:318]: same here

File: chef/chef.py [hunk 3]
```diff
@@ -233,34 +299,34 @@ def update_directory(self, method, local_dir, has_remote, branch, rev):
...
 
-        os.makedirs(path, exist_ok=True)
+        os.makedirs(target.build_dir(), exist_ok=True)
 
-        conf_path = os.path.join(path, 'conf')
+        conf_path = os.path.join(target.build_dir(), 'conf')
         os.makedirs(conf_path, exist_ok=True)
 
-        dl_dir = '${TOPDIR}/' + os.path.relpath(self.config.dl_dir(), path)
+        dl_dir = '${TOPDIR}/' + os.path.relpath(self.config.dl_dir(), target.build_dir())
```
Comment (adri1mart1) [inline chef/chef.py:315]: same here

File: chef/chef.py (hunk 1 above)
Comment (pboettch) [reply chef/chef.py:312]: OK.

File: chef/chef.py (hunk 2 above)
Comment (pboettch) [reply chef/chef.py:318]: I don't think so, only the build-dir of the build-config is used here, the rest is related to the command.

Maybe we should move the whole function to the class.

**Comment by cpb-** [comment]
To be included in #50

**Pull request closed without merging**